texastribune/ds-toolbox
# Restructure folders, deprecate some grid classes, update readme

#### What's this PR do?

- Adds a `7-layouts` folder to the mix
- Fixes some inconsistent naming and removes "Experimental" designation
- Adds markdown parser to CSS comment descriptions
- Breaks typography classes into individual files, (but keeps the helpers together)
- Adds a new layout to the homepage of of our docs site.
- `grid_container`=> `l-container`
- `grid_separator` => Already accounted for with `has-<size>-btm-marg` helper
- Brain dump in .README

#### Why are we doing this? How does it help us?

Breaking up layouts and utilities allows us to better visually parse the classes of our toolbox. As we start to commit this library to memory, I think the breakdown creates a better connection as to what lives where and the helpers at our disposal.

The other changes were just general housekeeping

#### How should this be manually tested?

Spin it up locally and make sure it works. There's a lot to take in on this one so overall, just ensure the organization makes sense, nothing looks wonky on the docs site, and all the naming feels right.

There's also a preview link [here](https://ttds-preview.surge.sh/index.html).

#### Are there any smells or added technical debt to note?

Yes. We'll have to combine this with some class name and `@include..." file name changes where we've used the new system already. That includes templates for the donor-wall, sponsor-wall, and podcasts. Also the UMP CSS might need some path/naming tweaks.

#### TODOs / next steps:

* [x] *Sort out these changes with where their instances on prod /texastribune* This is ready on a [branch](https://github.com/texastribune/texastribune/compare/ds-updates), but I don't want to cause any more conflicts for need for speed. Letting this whole reorg wait until after that's in.
